controllers/tortoise_controller_test.go

```diff
@@ -291,7 +313,222 @@ var _ = Describe("Test TortoiseController", func() {
 			}).Should(Succeed())
 		})
 		It("TortoisePhaseEmergency", func() {
-			// TODO: add
+			now := time.Now()
+			tc := testCase{
+				before: resources{
+					tortoise: utils.NewTortoiseBuilder().
+						SetName("mercari").
+						SetNamespace("default").
+						SetTargetRefs(v1alpha1.TargetRefs{
+							DeploymentName:              "mercari-app",
+							HorizontalPodAutoscalerName: pointer.String("hpa"),
+						}).
+						SetUpdateMode(v1alpha1.UpdateModeEmergency).
+						AddResourcePolicy(v1alpha1.ContainerResourcePolicy{
+							ContainerName: "app",
+							AutoscalingPolicy: map[corev1.ResourceName]v1alpha1.AutoscalingType{
+								corev1.ResourceCPU:    v1alpha1.AutoscalingTypeHorizontal,
+								corev1.ResourceMemory: v1alpha1.AutoscalingTypeVertical,
+							},
+						}).
+						SetTortoisePhase(v1alpha1.TortoisePhaseWorking). // will be updated.
+						SetRecommendations(v1alpha1.Recommendations{
+							Horizontal: &v1alpha1.HorizontalRecommendations{
+								TargetUtilizations: []v1alpha1.HPATargetUtilizationRecommendationPerContainer{
+									{
+										ContainerName: "app",
+										TargetUtilization: map[corev1.ResourceName]int32{
+											corev1.ResourceCPU: 50, // will be updated.
+										},
+									},
+								},
+								MaxReplicas: []v1alpha1.ReplicasRecommendation{
+									{
+										From:      0,
+										To:        24,
+										WeekDay:   now.Weekday().String(),
+										TimeZone:  now.Location().String(),
+										Value:     15, // will be updated
+										UpdatedAt: metav1.NewTime(now),
+									},
+								},
+								MinReplicas: []v1alpha1.ReplicasRecommendation{
+									{
+										From:      0,
+										To:        24,
+										WeekDay:   now.Weekday().String(),
+										TimeZone:  now.Location().String(),
+										Value:     3, // will be updated
+										UpdatedAt: metav1.NewTime(now),
+									},
+								},
+							},
+						}).
+						AddCondition(v1alpha1.ContainerRecommendationFromVPA{
+							ContainerName: "app",
+							Recommendation: map[corev1.ResourceName]v1alpha1.ResourceQuantity{
+								corev1.ResourceCPU:    {},
+								corev1.ResourceMemory: {},
+							},
+							MaxRecommendation: map[corev1.ResourceName]v1alpha1.ResourceQuantity{
+								corev1.ResourceCPU:    {},
+								corev1.ResourceMemory: {},
+							},
+						}).
+						Build(),
+					deployment: deploymentWithReplicaNum(10),
+				},
+			}
+
+			err := tc.createResources(ctx, k8sClient, cfg)
+			Expect(err).ShouldNot(HaveOccurred())
```

Comment:
This step will fail because VPA and HPA already exists. I added `cleanUP()` but even after deleting `mercari` tortoise the child VPA and HPA still exists. Is this expected?

Reply:
Ah, I completely forgot to add the cleanup, thanks.

> even after deleting mercari tortoise the child VPA and HPA still exists.

Yes, it's the expected behavior at the moment. Currently the controller doesn't delete VPA or HPA managed by tortoises when we delete tortoises. 

I think at least we can delete VPAs as you expected because the controller creates the VPAs when tortoise is created. 
But, for HPA, there're a possibility that HPA is created by the users and attached to tortoise by them/ So probably we don't want to delete it in that case.

Anyway, I'll create the issue, thanks for bringing this up. 👍

Reply:
https://github.com/mercari/tortoise/issues/58

Reply:
For now, please just delete all VPAs and HPAs in `cleanUp` fn 🙏

Reply:
Added - https://github.com/mercari/tortoise/pull/57/commits/fa6fa12ab0f89b568f63ce408324a044fe3928e6